src/main/java/org/mifos/connector/mpesa/camel/routes/SafaricomRoutesBuilder.java

```diff
@@ -0,0 +1,139 @@
+import org.apache.camel.Exchange;
+import org.apache.camel.LoggingLevel;
+import org.apache.camel.builder.RouteBuilder;
+import org.apache.camel.model.dataformat.JsonLibrary;
+import org.mifos.connector.mpesa.auth.AccessTokenStore;
+import org.mifos.connector.mpesa.dto.BuyGoodsPaymentRequestDTO;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+import java.util.Base64;
+import static org.mifos.connector.mpesa.camel.config.CamelProperties.ACCESS_TOKEN;
+import static org.mifos.connector.mpesa.camel.config.CamelProperties.BUY_GOODS_REQUEST_BODY;
+import static org.mifos.connector.mpesa.safaricom.config.SafaricomProperties.MPESA_BUY_GOODS_TRANSACTION_TYPE;
+
+
+@Component
+public class SafaricomRoutesBuilder extends RouteBuilder {
+
+    @Value("${mpesa.api.passKey}")
+    private String passKey;
+
+    @Value("${mpesa.api.lipana}")
+    private String buyGoodsBaseUrl;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @Autowired
+    private AccessTokenStore accessTokenStore;
+
+    private final Logger logger = LoggerFactory.getLogger(this.getClass());
+
+    @Override
+    public void configure() {
+
+        /*
+          Rest endpoint to initiate payment for buy goods
+
+          Sample request body: {
+              "BusinessShortCode": 174379,
+              "Amount": 1,
+              "PartyA": 254708374149,
+              "PartyB": 174379,
+              "PhoneNumber": 254708374149,
+              "CallBackURL": "https://mydomain.com/path",
+              "AccountReference": "CompanyXLTD",
+              "TransactionDesc": "Payment of X"
+            }
+         */
+        from("rest:POST:/buygoods")
+                .id("buy-goods-online")
+                .process(exchange -> {
+                    String body = exchange.getIn().getBody(String.class);
+                    BuyGoodsPaymentRequestDTO buyGoodsPaymentRequestDTO = objectMapper.readValue(
+                            body, BuyGoodsPaymentRequestDTO.class);
+
+                    exchange.setProperty(BUY_GOODS_REQUEST_BODY, buyGoodsPaymentRequestDTO);
+                    logger.info(body);
+                    logger.info(buyGoodsPaymentRequestDTO.toString());
+
+                })
+                .to("direct:buy-goods-base");
+
+        /*
+         * Starts the payment flow
+         *
+         * Step1: Authenticate the user by initiating [get-access-token] flow
+         * Step2: On successful [Step1], directs to [lipana-buy-goods] flow
+         */
+        from("direct:buy-goods-base")
+                .id("buy-goods-base")
+                .log(LoggingLevel.INFO, "Starting buy goods flow")
+                .to("direct:get-access-token")
+                .process(exchange -> {
+                    accessTokenStore.setExpiresOn(3599);
```

Comment:
Why are we setting expiry here? Also, please avoid hardcoding. 
We are already saving data in src/main/java/org/mifos/connector/mpesa/auth/AuthRoutes.java:L57

Reply:
My mistake! I did that for testing, Ill remove this.

Reply:
this is fixed